GraknOptions.java

```diff
@@ -57,6 +58,15 @@ public GraknOptions infer(boolean infer) {
         return this;
     }
 
+    public Optional<Boolean> inferLog() {
+        return Optional.ofNullable(inferLog);
+    }
+
+    public GraknOptions inferLog(boolean inferLog) {
+        this.inferLog = inferLog;
+        return this;
+    }
```

Comment:
are we sure this is the best name? not `traceInference` ? if we go with "log", I think it should at least be `logInference`?

Reply:
We first went with `inferenceLogging` and changed infer` -> `inference` for consistency, but @alexjpwalker pointed out that `inference` didn't look consistent with the other options. We then went with `inferLog` since from a user perspective this is clearly scoped to relate to the existing `infer` flag.
@alexjpwalker does `traceInference` sound good to you?

Reply:
`traceInference` definitely sounds nicer and is consistent with the other flags, but not with `infer` specifically. Hmmm.